Automatic update of searx/translations/messages.pot and messages.po files.

dalf opened the pull request:
## What does this PR do?

See commit messages:
* remove .tx directory (transifex)
* babel.cfg: ignore node_modules directories
* update searx_extra/update/update_translations.sh
   * the script expects searx/translations/messages.pot to be commited
   * it calls "pybabel extract", and if there is a meaningful change,
   * it calls "pybabel update"
* github workflow:
   * after each on the master branch of searxng (not elsewhere), a call `searx_extra/update/update_translations.sh` check if there is some changes on `searx/translations/messages.pot`
   * if yes it create a PR with a commit similar to 596577e907607b582709aa7afd51a0cc9df9ca9f so weblate can get the last messages.
* update the messages.po files (it haven't been done since a while).

The workflow is disabled on this branch, how it is on the master branch:
![image](https://user-images.githubusercontent.com/1594191/127747566-086f561e-0332-4aca-abd2-873baa3648d2.png)

see https://github.com/dalf/searxng/actions/runs/1085458894

If this github workflow doesn't create issue later the PR can be replaced by a commit on the master branch.

## Why is this change important?

Related to #195 / integration of weblate.

## How to test this PR locally?

* call  `searx_extra/update/update_translation.sh`
* call again `searx_extra/update/update_translation.sh`: see there is no change (no call to `pybabel update`)
* change `searx/translations/messages.pot`
* call again `searx_extra/update/update_translation.sh`: check that the script calls `pybabel update`

The github workflow can only be tested on the master branch, add a commit to replace `searxng`:
https://github.com/searxng/searxng/blob/53156cd4c66c5fe0a6059515f5df7b7bbe02d35a/.github/workflows/integration.yml#L96

## Author's checklist

<!-- additional notes for reviewiers -->

## Related issues

Related to #195

dalf committed:
babel.cfg: ignore node_modules directories

The NPM packages contains python scripts: they should be included by "pybabel extract"

dalf committed:
babel: update searx_extra/update/update_translations.sh

the script expects searx/translations/messages.pot to be commited

it calls "pybabel extract", and if there is a meaningful change,
it calls "pybabel update"

exit code 0 when there is a change in messages.pot

dalf committed:
remove .tx directory (transifex)

dalf committed:
github workflow: babel jobs to update messages.pot

dalf committed:
Update translations (pot, po)

return42 commented on .github/workflows/integration.yml:
```diff
@@ -90,6 +90,47 @@ jobs:
+      continue-on-error: true
+      run: |
+        pip install babel jinja2
+        searx_extra/update/update_translations.sh
```
This code should be executed in the virtualenv (./local/py3) / not in the environment from the OS.

dalf replied:
Note: the CI runs this code in a throw away VM: everything runs as root.

return42 replied:
Anyway, we do all build task in a dedicated environment and this is the environment in ./local/py3

By example: I tried to update translations and it fails on my desktop PC.

With commit 900baf2  all pybabel commands are executed in this environment by  the leading `pyenv.cmd pybabel`

return42 committed:
[mod] manage - implement babel commands, drop update_translations.sh

In ./manage implement babel.*:

-  extract   : extract messages from source files and generate POT file
-  update    : update existing message catalogs from POT file
-  compile   : compile translation catalogs into binary MO files

Replace searx_extra/update/update_translations.sh by command:

- ci.babel.update

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 commented on babel.cfg:
```diff
@@ -1,3 +1,4 @@
+[ignore: **/node_modules/**]
```
See my https://github.com/searxng/searxng/issues/141#issuecomment-890481065

return42 reviewed (approved):
LGTM .. lets merge.
